contracts/compounder/GenericVault.sol

```diff
@@ -0,0 +1,335 @@
+        ERC20(
+            string(abi.encodePacked("Staked ", ERC20(_token).name())),
+            string(abi.encodePacked("stk", ERC20(_token).symbol()))
+        )
+    {
+        underlying = _token;
+    }
+
+    /// @notice Updates the withdrawal penalty
+    /// @param _penalty - the amount of the new penalty (in BIPS)
+    function setWithdrawalPenalty(uint256 _penalty) external onlyOwner {
+        require(_penalty <= MAX_WITHDRAWAL_PENALTY);
+        withdrawalPenalty = _penalty;
+        emit WithdrawalPenaltyUpdated(_penalty);
+    }
+
+    /// @notice Set the address of the strategy contract
+    /// @dev Can only be set once
+    /// @param _strategy - address of the strategy contract
+    function setStrategy(address _strategy) external onlyOwner notToZeroAddress(_strategy) {
+        require(strategy == address(0), "Strategy already set");
+        strategy = _strategy;
+        emit StrategySet(_strategy);
+    }
+
+    /// @notice Count of extra rewards
+    function extraRewardsLength() external view returns (uint256) {
+        return extraRewards.length;
+    }
+
+    /// @notice Add extra reward contract
+    /// @param _reward VirtualBalanceRewardPool address
+    /// @return bool success
+    function addExtraReward(address _reward) external onlyOwner notToZeroAddress(_reward) returns (bool) {
+        if (extraRewards.length >= 12) {
+            return false;
+        }
+
+        extraRewards.push(_reward);
+        emit ExtraRewardAdded(_reward);
+        return true;
+    }
+
+    /// @notice Clear extra rewards array
+    function clearExtraRewards() external onlyOwner {
+        uint256 len = extraRewards.length;
+        for (uint256 i = 0; i < len; i++) {
+            emit ExtraRewardCleared(extraRewards[i]);
+        }
+        delete extraRewards;
+    }
+
+    /// @notice Query the amount currently staked
+    /// @return total - the total amount of tokens staked
+    function totalUnderlying() public view returns (uint256 total) {
+        return IStrategy(strategy).totalUnderlying();
+    }
+
+    /// @notice Returns the amount of underlying a user can claim
+    /// @param user - address whose claimable amount to query
+    /// @return amount - claimable amount
+    /// @dev Does not account for penalties and fees
+    function balanceOfUnderlying(address user) external view returns (uint256 amount) {
+        require(totalSupply() > 0, "No users");
+        return ((balanceOf(user) * totalUnderlying()) / totalSupply());
+    }
+
+    /// @notice Deposit user funds in the autocompounder and mints tokens
+    /// representing user's share of the pool in exchange
+    /// @param _amount - the amount of underlying to deposit
+    /// @return _shares - the amount of shares issued
+    function deposit(uint256 _amount, address _receiver)
+        public
+        notToZeroAddress(_receiver)
+        nonReentrant
+        returns (uint256 _shares)
+    {
+        require(_amount > 0, "Deposit too small");
```

Comment:
I think we should bump minimum deposit to 1000 wei or higher to help [catch the edge case](https://github.com/code-423n4/2022-11-redactedcartel-findings/issues/275).

Reply:
Yeah we can either add a minimum of 1e18 or something or we just do the initial deposit along with the deployment. So we can add a line in the deployment script to do this. If it gets front run which is unlikely we just re-deploy?

Reply:
The Latter option might be best. In support of changing the deploy script to include a deposit line. I'd support that initial deposit being sent to 0xdead too